drivers/console/console_private.h

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2019, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef CONSOLE_PRIVATE_H
+#define CONSOLE_PRIVATE_H
+
+#include <drivers/console.h>
+
+/* Private console_register() function. */
+int console_register(console_t *console);
```

Comment:
I think this should just go into console.h, there's no need to create a separate header. The "There is no publicly accessible console_register() function." comment in there is outdated -- since we support consoles written only in C now, those need to be able to call console_register().

Reply:
Can you provide a description of `console_register` to put above its declaration in console.h?

Reply:
"Add a console_t instance to the console list. This should only be called by console drivers after they have initialized all fields in the console structure. Platforms seeking to register a new console need to call the respective console_<driver>_register() function instead."